eppo_core/src/ufc/assignment.rs

```diff
@@ -37,6 +48,38 @@ pub enum AssignmentValue {
     },
 }
 
+impl Serialize for AssignmentValue {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut state = serializer.serialize_struct("AssignmentValue", 2)?;
+        match self {
+            AssignmentValue::String(s) => {
+                state.serialize_field("type", "STRING")?;
+                state.serialize_field("value", s)?;
+            }
+            AssignmentValue::Integer(i) => {
+                state.serialize_field("type", "INTEGER")?;
+                state.serialize_field("value", i)?;
+            }
+            AssignmentValue::Numeric(n) => {
+                state.serialize_field("type", "NUMERIC")?;
+                state.serialize_field("value", n)?;
+            }
+            AssignmentValue::Boolean(b) => {
+                state.serialize_field("type", "BOOLEAN")?;
+                state.serialize_field("value", b)?;
+            }
+            AssignmentValue::Json { raw: _, parsed } => {
+                state.serialize_field("type", "JSON")?;
+                state.serialize_field("value", parsed)?;
+            }
+        }
+        state.end()
+    }
+}
```

Comment:
This is the real fix. The rest is just moving away from magnus_serde.

I initially wanted to remove `Serialize` implementation completely and rely on Ruby-specific conversion functions. But that took quite a bit of work and we still need magnus_serde to translate Json's parsed value to Ruby, and it broke Rust's serialization of evaluation details.

So I just bit the bullet and implemented `Serialize` for `AssignmentValue`.